src/hypothesis/strategies.py

```diff
@@ -711,6 +711,29 @@ def text(
 
 @cacheable
 @defines_strategy
+def text_matching(regex):
```

Comment:
(see #212 for more examples, and keep bikeshedding!)

`strings_matching_regex` is longer than any current name, and feels too verbose to me.  I don't have a better suggestion than `from_regex` though.